### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by ElsonOtake**
- Validations added for the User model;
- Validations added for the Post model;
- Validations added for the Comment model;
- Unit specs added for all models' methods and validations.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@ class User < ApplicationRecord
   has_many :comments, foreign_key: 'author_id'
   has_many :likes, foreign_key: 'author_id'
   has_many :posts, foreign_key: 'author_id'
+  validates :name, presence: true
```
Comment (t-yanick) [inline app/models/user.rb:6]: - [ ] As it is a [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md) for this project, " `PostsCounter must be an integer greater than or equal to zero.` ", please kindly add validation rule for that. 

Please kindly ensure all custom methods in the models should have a corresponding unit test.

File: app/models/post.rb [hunk 2]
```diff
@@ -2,6 +2,9 @@ class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
   has_many :comments
   has_many :likes
+  validates :title, presence: true, length: { maximum: 250, too_long: '%<count>s characters is the maximum allowed' }
+
```
Comment (t-yanick) [inline app/models/post.rb:6]: - [ ] As it is a requirement for this project, please kindly add the following  validation rules to your `post model`. 
  - CommentsCounter must be an integer greater than or equal to zero.
  - LikesCounter must be an integer greater than or equal to zero. 

Please kindly ensure all custom methods in the models should have a corresponding unit test.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @ElsonOtake, 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark: Clean/Organized code
:heavy_check_mark: Descriptive Readme file

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

###  N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by t-yanick** [comment]
Hello @ElsonOtake, I mistakenly approved your project on my dashboard, please make sure you implement the requested changes.

**Comment by ElsonOtake** [comment]
> Hello @ElsonOtake, I mistakenly approved your project on my dashboard, please make sure you implement the requested changes.

@t-yanick Thanks. I will implement the requested changes. Have a nice day.

**Pull request merged**